pandas/io/sql.py

```diff
@@ -1011,7 +1011,11 @@ def _harmonize_columns(self, parse_dates=None):
 
                 elif len(df_col) == df_col.count():
                     # No NA values, can convert ints and bools
-                    if col_type is np.dtype("int64") or col_type is bool:
+                    if (
+                        col_type is np.dtype("int64")
+                        or col_type is np.dtype("int32")
```

Comment:
Does this need an extra consideration for int16?

Reply:
Thank you for the suggestion - I have included int16